### PR TITLE
add unit test for function ParseKindArg and ParseGroupKind

### PR DESCRIPTION
**Pull request by hanxiaoshuai**
**What this PR does / why we need it**:
add unit test for function ParseKindArg and ParseGroupKind
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go [hunk 1]
```diff
@@ -134,3 +134,44 @@ func TestKindForGroupVersionKinds(t *testing.T) {
...
+		}
+	}
+}
+
+func TestParseGroupKind(t *testing.T) {
+	tests := []struct {
+		input string
+		out   GroupKind
+	}{
+		{input: "v1", out: GroupKind{Kind: "v1"}},
```
Comment (hzxuzhonghu) [inline staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go]: kind should like `Pod` `StatefulSet`

File: staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go [hunk 2]
```diff
@@ -134,3 +134,44 @@ func TestKindForGroupVersionKinds(t *testing.T) {
...
+		{input: ".v1", out: GroupKind{Group: "v1"}},
```
Comment (hzxuzhonghu) [inline staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go]: group like `apps`

File: staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go [hunk 3]
```diff
@@ -134,3 +134,44 @@ func TestKindForGroupVersionKinds(t *testing.T) {
...
+		{input: "a.v1", out: GroupKind{Group: "v1", Kind: "a"}},
+		{input: "a.v1.b", out: GroupKind{Group: "b", Kind: "a.v1"}},
```
Comment (hzxuzhonghu) [inline staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go]: no such case

File: staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go (hunk 1 above)
Comment (hanxiaoshuai) [reply staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go]: Done

File: staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go (hunk 2 above)
Comment (hanxiaoshuai) [reply staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go]: Done

File: staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go (hunk 3 above)
Comment (hanxiaoshuai) [reply staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go]: Done

File: staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go [hunk 4]
```diff
@@ -134,3 +134,43 @@ func TestKindForGroupVersionKinds(t *testing.T) {
...
+		{input: "Pod", gk: GroupKind{Kind: "Pod"}},
+		{input: ".apps", gk: GroupKind{Group: "apps"}},
+		{input: "Pod.", gk: GroupKind{Kind: "Pod"}},
+		{input: "Pod.apps", gk: GroupKind{Group: "apps", Kind: "Pod"}},
+		{input: "Pod.v1.apps", gvk: &GroupVersionKind{Group: "apps", Version: "v1", Kind: "Pod"}, gk: GroupKind{Group: "v1.apps", Kind: "Pod"}},
+	}
+	for i, test := range tests {
+		gvk, gk := ParseKindArg(test.input)
+		if (gvk != nil && test.gvk == nil) || (gvk == nil && test.gvk != nil) || (test.gvk != nil && *gvk != *test.gvk) {
+			t.Errorf("%d: expected output: %#v, got: %#v", i, test.gvk, gvk)
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go]: Please use the subtest construct: https://blog.golang.org/subtests

**Comment by deads2k** [comment]
I like the test, I'd like them structured as subtests please.

File: staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go (hunk 4 above)
Comment (hanxiaoshuai) [reply staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go]: @deads2k Done

File: staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go [hunk 5]
```diff
@@ -134,3 +135,47 @@ func TestKindForGroupVersionKinds(t *testing.T) {
...
+		gk    GroupKind
+	}{
+		{input: "Pod", gk: GroupKind{Kind: "Pod"}},
+		{input: ".apps", gk: GroupKind{Group: "apps"}},
+		{input: "Pod.", gk: GroupKind{Kind: "Pod"}},
+		{input: "Pod.apps", gk: GroupKind{Group: "apps", Kind: "Pod"}},
+		{input: "Pod.v1.apps", gvk: &GroupVersionKind{Group: "apps", Version: "v1", Kind: "Pod"}, gk: GroupKind{Group: "v1.apps", Kind: "Pod"}},
+	}
+	for i, test := range tests {
+		t.Run(fmt.Sprintf("%s", test.input), func(t *testing.T) {
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go]: You should be able to just do `test.Run(test.input, func(***`, right? test.input is already a string.

**Comment by deads2k** [comment]
minor comment on unnecessary fmt.sprintf

/approve
/assign hzxuzhonghu

File: staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go [hunk 6]
```diff
@@ -134,3 +135,47 @@ func TestKindForGroupVersionKinds(t *testing.T) {
...
 	}
 }
+
+func TestParseKindArg(t *testing.T) {
+	tests := []struct {
+		input string
+		gvk   *GroupVersionKind
+		gk    GroupKind
+	}{
+		{input: "Pod", gk: GroupKind{Kind: "Pod"}},
```
Comment (hzxuzhonghu) [inline staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go]: nit:  pod not belong to apps, same below

**Pull request merged**